.github/workflows/build-proto-image.yml

```diff
@@ -3,13 +3,99 @@
 name: Proto Build Workflow
 
 on:
-  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+    tags:
+      # semver, e.g. 1.2.0 (does not match 0.1.2)
+      - "[1-9]+.[0-9]+.[0-9]+"
+      # semver with prerelease info, e.g. 1.0.2-beta.1 or 1.2.3-rc.10
+      - "[1-9]+.[0-9]+.[0-9]+-[a-z]+.[0-9]+"
+      # do not match prerelease starting w/ 0, e.g. 1.0.2-beta.0 or 1.2.3-rc.01
+      - "![1-9]+.[0-9]+.[0-9]+-[a-z]+.[0]*"
+      # semver with date info, e.g. 1.0.2-20221125
+      - "[1-9]+.[0-9]+.[0-9]+-[0-9]+"
+      # do not match date starting w/ 0, e.g. 1.0.2-01232023
+      - "![1-9]+.[0-9]+.[0-9]+-[0]*"
   pull_request:
     branches:
       - "main"
 
 jobs:
-  dummy:
+  build_proto_image:
+    name: Build Proto Image
     runs-on: ubuntu-latest
+
     steps:
-      - run: echo "Placeholder workflow"
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Create image metadata
+        id: meta
+        uses: ./.github/actions/metadata-action
+        with:
+          repository: axisecp/acap-runtime
+          get_version: "true"
+
+      - name: Build Proto image
+        uses: ./.github/actions/docker-build-push-action
+        with:
+          dockerfile: ./Dockerfile.proto
+          push: false
+          load: true
+          tags: axisecp/acap-runtime:${{ steps.meta.outputs.version }}-protofiles
+          use_qemu: true
+          platform: linux/arm64
+
+      - name: Extract proto files
+        run: |
+          container_id=$(docker create --platform linux/arm64 axisecp/acap-runtime:${{ steps.meta.outputs.version }}-protofiles)
+          docker cp $container_id:/build/param/proto_utils ./proto_utils_param
+          docker cp $container_id:/build/vdo/proto_utils ./proto_utils_vdo
+          docker cp $container_id:/build/tf/proto_utils ./proto_utils_tf
+          docker rm $container_id
+
+      - name: Verify proto files
```

Comment:
This step doesn't directly affect the final image, and removing it would streamline the workflow. However, this verification step serves as a safeguard against potential future changes that might break the image content.

By checking for the presence of the proto files before pushing, we can catch issues early and prevent incomplete or broken images from being published.